plugins/PrepareStage/PrepareMenu.qml

```diff
@@ -19,53 +19,102 @@ Item
         name: "cura"
     }
 
-    Row
+    // Item to ensure that all of the buttons are nicely centered.
+    Item
     {
         anchors.horizontalCenter: parent.horizontalCenter
+        width: openFileButtonBackground.width + itemRowBackground.width
+        height: parent.height
 
-        Button
+        Rectangle
         {
-            id: openFileButton
-            text: catalog.i18nc("@action:button", "Open File")
-            iconSource: UM.Theme.getIcon("load")
-            style: UM.Theme.styles.tool_button
-            tooltip: ""
-            action: Cura.Actions.open
-        }
+            id: itemRowBackground
+            radius: UM.Theme.getSize("default_radius").width
 
-        Item
-        {
-            id: spacing
-            width: UM.Theme.getSize("default_margin").width
-            height: prepareMenu.height
-        }
+            color: UM.Theme.getColor("toolbar_background")
 
-        Cura.MachineSelector
-        {
-            id: machineSelection
-            width: UM.Theme.getSize("machine_selector_widget").width - configSelection.width
-            height: prepareMenu.height
-            z: openFileButton.z - 1
-        }
+            width: itemRow.width + UM.Theme.getSize("default_margin").width
+            height: parent.height
 
-        Cura.QuickConfigurationSelector
-        {
-            id: configSelection
-            width: visible ? UM.Theme.getSize("machine_selector_widget").width * 0.2 : 0
-            panelWidth: UM.Theme.getSize("machine_selector_widget").width
-            height: prepareMenu.height
-        }
+            anchors.left: openFileButtonBackground.right
+            anchors.leftMargin: UM.Theme.getSize("default_margin").width
 
-        Cura.CustomConfigurationSelector
-        {
-            width: UM.Theme.getSize("configuration_selector_widget").width
+            RowLayout
+            {
+                id: itemRow
+
+                anchors.centerIn: parent
+
+                width: 0.9 * prepareMenu.width
+                height: parent.height
+                spacing: 0
+
+                Cura.MachineSelector
+                {
+                    id: machineSelection
+                    z: openFileButtonBackground.z - 1
```

Comment:
If this is to fix the hover text of the open file button falling behind the item row, I think it warrants a comment of documentation.
```suggestion
                    z: openFileButtonBackground.z - 1 //Ensure that the tooltip of the open file button stays above the item row.
```